src/components/shared/Navigation.tsx

```diff
@@ -272,73 +275,100 @@ const ProfileTab = createTabNavigator(
+      ),
+      navigationOptions: { header: null }
+    },
+    App: createMaterialBottomTabNavigator(
+      {
+        HomeTab,
+        DiscoverTab,
+        MintTab,
+        ProfileTab
+      },
+      {
+        initialRouteName: RouteName.DiscoverTab,
+        labeled: false,
+        navigationOptions: ({ navigation }: any) => ({
+          tabBarIcon: ({ focused }: any) => {
+            const { routeName } = navigation.state;
+            let iconName;
+            let IconType = Icon;
+            switch (routeName) {
+              case RouteName.ProfileTab:
+                iconName = "account";
+                break;
+              case RouteName.HomeTab:
+                iconName = "home";
+                break;
+              case RouteName.MintTab:
+                iconName = "gift";
+                break;
+              case RouteName.DiscoverTab:
+                iconName = "ios-search";
+                IconType = Ionicons;
+                break;
+              default:
+                throw Error(`Unrecognized route ${routeName}`);
+            }
+            const isMint = routeName === RouteName.MintTab;
+            if (!focused && !isMint) {
+              iconName += "-outline";
+            }
+            return <IconType name={iconName} size={25} />;
+          },
+          headerStyle: {
+            backgroundColor: colors.lightBackground
+          },
+          labelStyle: {
+            color: colors.bodyText
+          },
+          tabBarColor: colors.lightAccent
+        })
+      }
+    )
   },
   {
-    initialRouteName: RouteName.DiscoverTab,
-    labeled: false,
-    navigationOptions: ({ navigation }: any) => ({
-      tabBarIcon: ({ focused }: any) => {
-        const { routeName } = navigation.state;
-        let iconName;
-        let IconType = Icon;
-        switch (routeName) {
-          case RouteName.ProfileTab:
-            iconName = "account";
-            break;
-          case RouteName.HomeTab:
-            iconName = "home";
-            break;
-          case RouteName.MintTab:
-            iconName = "gift";
-            break;
-          case RouteName.DiscoverTab:
-            iconName = "ios-search";
-            IconType = Ionicons;
-            break;
-          default:
-            throw Error(`Unrecognized route ${routeName}`);
-        }
-        const isMint = routeName === RouteName.MintTab;
-        if (!focused && !isMint) {
-          iconName += "-outline";
-        }
-        return <IconType name={iconName} size={25} />;
-      },
-      headerStyle: {
-        backgroundColor: colors.lightBackground
-      },
-      labelStyle: {
-        color: colors.bodyText
-      },
-      tabBarColor: colors.lightAccent
-    })
+    initialRouteName: RouteName.LoadingSplash
   }
 );
 
-const SignedOutNavigator = createStackNavigator(
+const SignedOutNavigator = createSwitchNavigator(
   {
-    Onboarding: {
-      screen: Onboarding,
+    [RouteName.LoadingSplash]: {
```

Comment:
Why is this an array?

Reply:
Ah, it's kind of weird syntax but it's actually dereferencing LoadingSplash from RouteName and using that as the value of the key -- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names